sklearn/isotonic.py

```diff
@@ -190,6 +190,12 @@ class IsotonicRegression(RegressorMixin, TransformerMixin, BaseEstimator):
     X_max_ : float
         Maximum value of input array `X_` for right bound.
 
+    X_thresholds_ : array-like of shape (n_thresholds,)
+        De-duplicated ascending `X` values suitable to interpolate the y = f(X) monotonic function.
```

Comment:
I discovered in https://github.com/scikit-learn/scikit-learn/issues/16321#issuecomment-580695674 that the X thresholds are actually not exactly ascending. I am not 100% sure if this is intentional or not. Will need to investigate but writing a test to check this property will require a fix.

Reply:
This was wrong. I added a new test as part of this PR and `X_threholds_` is always monotonic (on all the random data I tried).